File: MarcenariaApi/Models/Tarefa.cs
using MarcenariaApi.Models.Enum;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace MarcenariaApi.Models
{
    public class Tarefa
    {
        [Key]
        public int? id { get; set; }
        public string? nome { get; set; }
        public string? desc { get; set; }
        public Status status { get; set; }
        public DateTime dataInicio { get; set; }
        public DateTime? dataFinalizacao { get; set; }

        public int ProjetoId { get; set; }

        [JsonIgnore]
        public Projeto? Projeto { get; }
    }
}

File: AppMarcenaria/src/app/models/tarefas/Tarefa.ts
import { Status } from '../status/Status';
import { Projeto } from '../projetos/Projeto';
import { Material } from '../materiais/Material';

export interface Tarefa {
    id: number;
    nome: string;
    desc: string;
    status: Status;
    dataInicio: Date;
    dataFinalizacao: Date;
    projetoId: number;
    projeto?: Projeto;
}
